src/com/amplitude/api/AmplitudeClient.java

```diff
@@ -320,6 +322,50 @@ public AmplitudeClient enableNewDeviceIdPerInstall(boolean newDeviceIdPerInstall
     }
 
     /**
+     * Whether to enable diagnostic logging. If true, then the SDK will log diagnostic events
+     * for upload failures to a separate Diagnostic Event endpoint.
+     * @param enableDiagnosticLogging whether to enable diagnostic logging. If false then disable.
+     * @return the AmplitudeClient
+     */
+    public AmplitudeClient enableDiagnosticLogging(boolean enableDiagnosticLogging) {
+        if (!contextAndApiKeySet("enableDiagnosticLogging")) {
+            return this;
+        }
+
+        this.logDiagnosticEvents = enableDiagnosticLogging;
+        if (enableDiagnosticLogging) {
+            Diagnostics.getLogger(context).enableLogging(httpClient, apiKey);
```

Comment:
this actually needs to be run on the logThread to make sure initialize finishes running, otherwise httpClient will be null even though there might be an apikey